overlord/overlord.go

```diff
@@ -78,6 +78,7 @@ func New() (*Overlord, error) {
 	}
 	o.snapMgr = snapMgr
 	o.stateEng.AddManager(o.snapMgr)
+	snapMgr.Init(s)
```

Comment:
What's the deal here?  Have we found a case where we need the state earlier than we expected?

If necessary, I think it's fine to drop Init altogether and have Manager(state *State) instead.